File: question4.c
#include <stdio.h>

void moveZerosToEnd(int arr[], int n) {
    int count = 0;  // Count of non-zero elements

    // Move non-zero elements to the front
    for (int i = 0; i < n; i++) {
        if (arr[i] != 0) {
            arr[count++] = arr[i];
        }
    }

    // Fill remaining positions with 0
    while (count < n) {
        arr[count++] = 0;
    }
}

int main() {
    int arr[] = {0, 1, 0, 3, 12};
    int n = sizeof(arr) / sizeof(arr[0]);

    moveZerosToEnd(arr, n);

    printf("Array after moving zeros to end:\n");
    for (int i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }

    return 0;
}

File: question4.py
arr = [0, 1, 0, 3, 12]

# Two-pointer approach
count = 0  # Count of non-zero elements

for i in range(len(arr)):
    if arr[i] != 0:
        arr[count] = arr[i]
        count += 1

# Fill the rest with zeros
while count < len(arr):
    arr[count] = 0
    count += 1

print("Array after moving zeros to end:", arr)
